torchaudio_contrib/functional.py

```diff
@@ -47,6 +47,10 @@ def stft(signal, fft_len, hop_len, window,
     return spect
 
 
+def spectral_downmix(tensor, power=1.0):
+    return torch.sum(tensor**2/power, 1, keepdim=True) ** (power/2)
+
```

Comment:
Wouldn't it be `tensor**(2/power)`? I've done some timing and `torch.norm(z, p=2, dim=1, keepdim=True)` is faster for `cuda`, although then I guess the meaning of `power` changes.

Reply:
+1. 

I've also seen some folks talk about quite big differences of execution time between (something like) those. Not sure it'd be significant for us though.

Reply:
I assume `tensor.shape=(batch, channel, time and freq in the right order)` as we discussed before? The name `tensor` sounds a bit too general, I can't think of a great name though. But it'd be useful in other places as well if we coin some names for waveform batch, spectrogram batch, etc. ...Oops maybe `waveform_batch` and `spectrogram_batch`?

Reply:
Yeah [like](https://github.com/pytorch/pytorch/issues/5671) [these](https://github.com/pytorch/pytorch/pull/5722). Using `torch.autograd.profiler` I always find `torch.norm` faster than manual on cuda but slower on cpu...

Reply:
> I always find `torch.norm` faster than manual on cuda but slower on cpu...

We should still prefer `torch.norm` then, to benefit from possible future optimizations of that. If we opt for a particular manual formulation, then there's nothing the PyTorch developers can do to speed it up.

Reply:
And why should we use the power rather than the mean?

Reply:
> And why should we use the power rather than the mean?

mean would also work, sure.

Reply:
I replaced it with mean (which also makes more sense)